services/src/main/scala/cromwell/pipeline/service/ProjectService.scala

```diff
@@ -34,11 +34,36 @@ class ProjectService(projectRepository: ProjectRepository)(implicit executionCon
     projectRepository.addProject(project)
   }
 
-  def deactivateProjectById(projectId: ProjectId): Future[Option[Project]] =
-    for {
-      _ <- projectRepository.deactivateProjectById(projectId)
-      getProject <- projectRepository.getProjectById(projectId)
-    } yield getProject
+  def deactivateProjectById(projectId: ProjectId, userId: UserId): Future[Option[Project]] = {
+    val result = projectRepository.getProjectById(projectId)
+    result.flatMap {
+      case Some(project) if project.ownerId != userId => throw new ProjectAccessDeniedException
+      case Some(_) =>
+        for {
+          _ <- projectRepository.deactivateProjectById(projectId)
+          getProject <- projectRepository.getProjectById(projectId)
+        } yield getProject
+      case None => throw new ProjectNotFoundException
+    }
+  }
+
+  def updateProject(request: ProjectUpdateRequest, userId: UserId): Future[Int] =
```

Comment:
Why does it return an `Int`? What does this `Int` mean?

Reply:
This Int - is Id of updated project

Reply:
1) We have a wrapper for the `ProjectId`. If you want to return the `Id` of the updated project, you must use this wrapper.
2) `ProjectId` is a wrapper for a `String`. Therefore, this `Int` is definitely not an `Id` of the updated project.

Reply:
Sorry, I was wrong. This is number of updated rows. Should I change this to a more convenient format? If so, which one?

Reply:
This is debatable.
You can either return `Future[Unit]` or `Future[Project]`. `Future[Unit]` returns empty `Success` if the project was updated successfully and `Failure` otherwise. `Future[Project]` is similar but instead of empty `Success`, it returns an updated project. Since during the deactivation of a project you are returning the deactivated project back to the user, I think it would be convenient to return the updated project here.
@ilyaonishenko @mnesvit What do you think?

Reply:
I agree with @myazinn

Reply:
fixed